eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/charsequence/Trimming.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.charsequence;
+
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+
+
+/**
+ * Trimming {@link CharSequence} decorator.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class Trimming extends AbstractCharSequence
```

Comment:
It was `Trimmed` first but I renamed to follow the decorator naming, like the `Following` or `UrlRewriting`. But `Trimmed` feels better to me, too, so I renamed it now.

It's not used, that's true, it just looked like something useful, so I kept it. Let me know if I should remove though.

About `LeadingEmptyLinesAndTrailTrimming` - see comment there.